### PR TITLE
Fix fstrings being escaped improperly

### PR DESCRIPTION
**Pull request by invrainbow**
### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

Previously fstrings were being improperly escaped, eg:

```python
f'asdf {rx.cond(blah, 'a', 'b')}'
```

would produce the JS:

```jsx
{`asdf ${ifTrue(blah) ? \`a\` : \`b\`}`}
```

This PR fixes interpolation handling to only escape the non-interpolated portions. Now it correctly produces

```jsx
{`asdf ${ifTrue(blah) ? `a` : `b`}`}
```

### CODE REVIEW

**Comment by invrainbow** [comment]
this contains commits from main that aren't merged into reflex-0.4.0 yet so waiting on those before merge

File: reflex/vars.py
Comment (masenf) [inline reflex/vars.py:264]:
```suggestion
            data = json_loads(m.group(1))
```

**Review by masenf — COMMENTED** [review]
this is cool, way to get your hands dirty in some of the yuckiest code in the framework 😅

**Comment by invrainbow** [comment]
@masenf made requested changes!

**Review by masenf — APPROVED** [review]

**Pull request merged**
